### ISSUE TITLE
Upgrade pytorch version to 2.0

### ISSUE THREAD

**Issue by Arnav0400** [transmuteAI/trailmet#148]
Since Pytorch 2.0 supports a variety of features, we need to test and upgrade to the same.

**Comment by sydarb** [comment]
true-quantization feature (#150) requires modules (`torch.ao.quantization`) available only in pytorch 2.0+ and all existing quantization methods have been tested for their compatibility with it.
